packages/core/src/components/fields/SchemaField.js

```diff
@@ -71,17 +71,41 @@ function Label(props) {
   );
 }
 
-function LabelInput(props) {
-  const { id, label, onChange } = props;
-  return (
-    <input
-      className="form-control"
-      type="text"
-      id={id}
-      onBlur={event => onChange(event.target.value)}
-      defaultValue={label}
-    />
-  );
+class LabelInput extends React.Component {
+  constructor(props) {
+    super();
+    this.state = {
+      value: props.label,
+    };
+  }
+  render() {
+    const { id, label, onChange, schema, registry } = this.props;
+
+    if (!schema) {
+      return (
+        <input
```

Comment:
I did not update this `<input />` as it seems to be a cause of breaking change if we replace that with `{ type: "string" }`.

Reply:
Sounds good. I'm curious, what would the breaking change have looked like? If we have replaced it with a SchemaField with type: string, I would have expected it to look the same, though perhaps there's something I'm missing.

Reply:
What I tried is the below SchemaField and faced 33 fails([detail](https://gist.github.com/watilde/9d731026a6f3edd5ba3df6b71c0b3fc0)).

```jsx
 <SchemaField
  schema={{type: "string"}}
  value={label}
  registry={registry}
  onChange={onChange}
  onBlur={(_, value) => onChange(value)}
/>;
```

Reply:
@epicfaace Could we replace the input with SchemaField separately from this PR? This PR uses the exactly same `<input>` tag with the current implementation.

Reply:
That seems OK. However, I'm just worried now that the `SchemaField` we _do_ implement in this PR (in the screenshot below) might also have similar bugs that we're not aware of:

![image](https://user-images.githubusercontent.com/1689183/133928580-4d1a20c4-dfd1-4301-bac7-74569059ccc5.png)

Can we try to get to the bottom of this? At least if we can understand why the <SchemaField> for the type: string was failing (and demonstrate that it's a different situation from the SchemaField we added, which wouldn't fail), then I'd be more comfortable merging this PR>

Reply:
@epicfaace Sure! I'll dig into that a bit more, but it'd be great if we do not need to refactor `SchemaField` when we find any. My ask was to keep the current behaviour for no schema case so that we can minimize breaking change risk as less as we can. As the screenshot you put shows no schema case is using the exactly same `<input />` tag with the current master, I'm wondering if we could discuss in a separated issue as the original issue is not related to that.

I'll investigate anyway.

Reply:
Turned out SchemaField puts wrong id. In the below example, it should be `id="root_first-key", but it uses `id="root"`. 
https://gist.github.com/watilde/bbd85528c7ba684e366ea451145c5f72#file-after-html-L10

Reply:
@epicfaace Ok, I fixed the issue by passing id as `idSchema`. We use only SchemaField now :)
https://github.com/rjsf-team/react-jsonschema-form/pull/2544/commits/72dcbf75d313e84bb776ec05238f30f7a8c8b771